File: src/main/java/br/com/season/controller/UserController.java
package br.com.season.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.util.UriComponentsBuilder;

import br.com.season.model.User;
import br.com.season.repositories.UserRepository;

@RestController
public class UserController {

	@Autowired
	UserRepository userRepository;
	
	@RequestMapping(value = "/user", method = RequestMethod.GET)
	public ResponseEntity<List<User>> listAllUsers() {
		List<User> users = userRepository.findAll();
		if(users.isEmpty()){
            return new ResponseEntity<List<User>>(HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<List<User>>(users, HttpStatus.OK);
	}
	
	@RequestMapping(value = "/user/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<User> getUser(@PathVariable("id") Integer id) {
        System.out.println("Fetching User with id " + id);
        User user = userRepository.findOne(id);
        if (user == null) {
            System.out.println("User with id " + id + " not found");
            return new ResponseEntity<User>(HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<User>(user, HttpStatus.OK);
    }
	
	@RequestMapping(value = "/user/", method = RequestMethod.POST)
    public ResponseEntity<Void> createUser(@RequestBody User user,    UriComponentsBuilder ucBuilder) {
        System.out.println("Creating User " + user.getName());
  
        if (userRepository.isUserExist(user.getName())) {
            System.out.println("A User with name " + user.getName() + " already exist");
            return new ResponseEntity<Void>(HttpStatus.CONFLICT);
        }
  
        userRepository.save(user);
  
        HttpHeaders headers = new HttpHeaders();
        headers.setLocation(ucBuilder.path("/user/{id}").buildAndExpand(user.getId()).toUri());
        return new ResponseEntity<Void>(headers, HttpStatus.CREATED);
    }

	@RequestMapping(value = "/user/{id}", method = RequestMethod.PUT)
    public ResponseEntity<User> updateUser(@PathVariable("id") Integer id, @RequestBody User user) {
        System.out.println("Updating User " + id);
          
        User currentUser = userRepository.findOne(id);
          
        if (currentUser==null) {
            System.out.println("User with id " + id + " not found");
            return new ResponseEntity<User>(HttpStatus.NOT_FOUND);
        }
  
        currentUser.setName(user.getName());
        currentUser.setAddress(user.getAddress());
        currentUser.setEmail(user.getEmail());
          
        userRepository.save(currentUser);
        return new ResponseEntity<User>(currentUser, HttpStatus.OK);
    }
	
	@RequestMapping(value = "/user/{id}", method = RequestMethod.DELETE)
    public ResponseEntity<User> deleteUser(@PathVariable("id") Integer id) {
        System.out.println("Fetching & Deleting User with id " + id);
  
        User user = userRepository.findOne(id);
        if (user == null) {
            System.out.println("Unable to delete. User with id " + id + " not found");
            return new ResponseEntity<User>(HttpStatus.NOT_FOUND);
        }
  
        userRepository.delete(id);
        return new ResponseEntity<User>(HttpStatus.NO_CONTENT);
    }
}

File: src/main/webapp/js/controller/user.controller.js
app.controller('userController', [
    '$scope', 'userService',
    function ($scope, userService) {
        var vm = this;
        vm.users = [];
        vm.user={id:null, name:'',address:'',email:''};
     
        vm.submit = submit;
        vm.edit = edit;
        vm.remove = remove;
        vm.reset = reset;
        
        listAllUsers();
        function listAllUsers(){
        		userService.listAllUsers()
                .then(
                function(d) {
                    vm.users = d;
                },
                function(errResponse){
                    console.error('Error while fetching Users');
                }
            );
        }
        
        function createUser(user){
            userService.createUser(user)
                .then(
                		listAllUsers,
                function(errResponse){
                    console.error('Error while creating User');
                }
            );
        }
        
        function updateUser(user, id){
        		userService.updateUser(user, id)
                .then(
                		listAllUsers,
                function(errResponse){
                    console.error('Error while updating User');
                }
            );
        }
        
        function deleteUser(id){
            userService.deleteUser(id)
                .then(
                		listAllUsers,
                function(errResponse){
                    console.error('Error while deleting User');
                }
            );
        }
     
        function submit() {
            if(vm.user.id===null){
                console.log('Saving New User', vm.user);
                createUser(vm.user);
            }else{
                updateUser(vm.user, vm.user.id);
                console.log('User updated with id ', vm.user.id);
            }
            reset();
        }
     
        function edit(id){
            console.log('id to be edited', id);
            for(var i = 0; i < vm.users.length; i++){
                if(vm.users[i].id === id) {
                    vm.user = angular.copy(vm.users[i]);
                    break;
                }
            }
        }
     
        function remove(id){
            console.log('id to be deleted', id);
            if(vm.user.id === id) {
                reset();
            }
            deleteUser(id);
        }
     
        function reset(){
            vm.user={id:null, name:'', address:'', email:''};
            $scope.myForm.$setPristine();
        }
    }
]);